pallets/pallet-bonded-coins/src/mock.rs

```diff
@@ -45,15 +46,30 @@ pub(crate) const DEFAULT_BONDED_DENOMINATION: u8 = 10;
 pub(crate) const ONE_HUNDRED_KILT: u128 = 100_000_000_000_000_000;
 
 // helper functions
-pub fn assert_relative_eq(target: Float, expected: Float, epsilon: Float) {
+pub fn assert_relative_eq<T: FixedSigned>(target: T, expected: T, epsilon: T) {
 	assert!(
 		(target - expected).abs() <= epsilon,
-		"Expected {:?} but got {:?}",
+		"Expected {:?} +/- {:?} but got {:?}",
 		expected,
+		epsilon,
 		target
 	);
 }
 
+pub(crate) fn collateral_at_supply(supply: u128) -> u128 {
```

Comment:
I would get rid of this function. I explained why in:

Reply:
It always needs to be used with `assert_eq_error_rate!` of course

Reply:
Could we then maybe rename the function? It is not really clear, that this function is calculating the collateral for the default linear bonding curve. Or make it part of the return type for the function, where the curve is created, but this is maybe a bit overengineered.

Reply:
41c70f501425b49dc750349b8b6f62084569c5c2